File: common/src/main/java/kg/delletenebre/yamus/api/response/Mix.kt
package kg.delletenebre.yamus.api.response

import kotlinx.serialization.Serializable

@Serializable
data class Mix(
    val data: Data,
    val id: String,
    val type: String
) {
    @Serializable
    data class Data(
        val backgroundColor: String,
        val backgroundImageUri: String,
        val coverWhite: String,
        val textColor: String,
        val title: String,
        val url: String,
        val urlScheme: String
    )
}

File: lib/api/models/mix.dart
class Mix {
  Mix({
    this.id = '',
    this.type = '',
    this.backgroundImageUri = '',
    this.title = '',
    this.url = '',
    this.urlScheme = '',
  });

  final String id;
  final String type;
  final String backgroundImageUri;
  final String title;
  final String url;
  final String urlScheme;
  
  factory Mix.fromJson(Map<String, dynamic> json) {
    return Mix(
      id: json['id'],
      type: json['type'],
      backgroundImageUri: json['data']['backgroundImageUri'],
      title: json['data']['title'],
      url: json['data']['url'],
      urlScheme: json['data']['urlScheme'],
    );
  }
}